colossalai/inference/core/request_handler.py

```diff
@@ -336,10 +336,20 @@ def search_tokens(self, generation_config: GenerationConfig, logits):
         Sample tokens for finished requests.
         """
 
+        # NOTE: need to decide the granularity to process logits (sequence or batch)
+        config_dict = generation_config.to_dict()
+        # process repetition_penalty, no_repeat_ngram_size
+        for type in ["repetition_penalty", "no_repeat_ngram_size"]:
+            if type in config_dict and config_dict[type] is not None:
+                if not self.prefill_bb.is_empty:
+                    batch = self.prefill_bb
+                else:
+                    batch = self.running_bb
+                logits = logit_processor(type, logits, config_dict[type], batch)
```

Comment:
Pass the BatchBucket here. In this way, redundant operations of converting lists to torch.tensors can be avoided, through numerical judgments within the logit_processor.

Reply:
Sure